### PR TITLE
Fix script + UnitTests (pytest) + CI (Github-Action)

### PR DESCRIPTION
**Pull request by yoyonel**
- Fixing import for using ModernGL in scripts.
- Adding unit tests with Pytest + xvfb + fakefilesystem and simple images comparaison.
- Adding a very simple CI with github-action (using flake8 and pytest).

### CODE REVIEW

**Comment by einarf** [comment]
I got access to this repo now. Will look at it very soon.

**Comment by yoyonel** [comment]
To integrate this PR, you have to change the badge url in README.md to link to this repo (not mine :p)

File: README.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+![](https://github.com/yoyonel/Headless-rendering-with-python/workflows/Python%20application/badge.svg)
```
Comment (yoyonel) [inline README.md:1]: Here, need to change the url (yoyonel <-> cprogrammer1994)

File: requirements.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
 Pillow
-ModernGL
+moderngl==5.5.4
```
Comment (yoyonel) [inline requirements.txt]: here we have to do something more generic, dynamic ... maybe a grid of versions, and multiple build targets ... something like this ... dunno :p

Comment (einarf) [reply requirements.txt]: `moderngl<6` should work for now

Comment (yoyonel) [reply requirements.txt]: image
Seems not to work :/

Comment (einarf) [reply requirements.txt]: Did this work with 5.5.4 or something? The only difference is that we're loading libGL dynamically

Comment (yoyonel) [reply requirements.txt]: Yep you can see on the CI working with `moderngl<5.6.0` at: https://github.com/yoyonel/Headless-rendering-with-python/runs/422066909?check_suite_focus=true
After merging (accept this PR), this repo will have the capacity to run their own CI (github-action pipeline).

Comment (yoyonel) [reply requirements.txt]: Oki i found the problem, in the container (github-action container) there is no link (symbolic) to `libGL.so` ... MmMmm .. i will add a symbolic link into the pipeline, i don't know if it's a good practice ... i have to search more.

Comment (einarf) [reply requirements.txt]: `ctx.create_context(..)` also supports passing in the `libgl='libGL.so.1'` if overrides are needed. Whatever works ;)

Comment (yoyonel) [reply requirements.txt]: image
yeahhhhhhh ^^
=> work with `moderngl==5.6.0`

**Comment by einarf** [comment]
I'm playing with a simple guide for Ubuntu 18 server as well using egl or x11 (and xvfb). Messing around in a VM atm.

**Comment by einarf** [comment]
Thank you. I'll tweak the remaining things.

**Pull request merged**